sql/core/src/test/scala/org/apache/spark/sql/DataFrameSuite.scala

```diff
@@ -2555,6 +2555,15 @@ class DataFrameSuite extends QueryTest
     val df = Seq(0.0 -> -0.0).toDF("pos", "neg")
     checkAnswer(df.select($"pos" > $"neg"), Row(false))
   }
+
+  test("SPARK-32816: aggregating multiple distinct DECIMAL columns") {
+    spark.range(0, 100, 1, 1)
+      .selectExpr("id", "cast(id as decimal(9, 0)) as decimal_col")
+      .createOrReplaceTempView("test_table")
```

Comment:
nit: wrap the test with `withTempView`

Reply:
How about writing it like this w/o a temp view;
```
  test("SPARK-32816: aggregating multiple distinct DECIMAL columns") {
    checkAnswer(
      sql(
        s"""
           |SELECT AVG(DISTINCT decimal_col), SUM(DISTINCT decimal_col)
           |  FROM VALUES (CAST(1 AS DECIMAL(9, 0))) t(decimal_col)
         """.stripMargin),
      Row(XXX, XXX))
  }
```

Reply:
nit: Also, could you move this test into `SQLQueryTestSuite`?

Reply:
let me try this, I'm not sure if literal will behave differently